src/webserver.cpp

```diff
@@ -0,0 +1,146 @@
+//    it under the terms of the GNU General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//   
+//    NightDriver is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU General Public License for more details.
+//   
+//    You should have received a copy of the GNU General Public License
+//    along with Nightdriver.  It is normally found in copying.txt
+//    If not, see <https://www.gnu.org/licenses/>.
+//
+// Description:
+//
+//   Implementations for some of the web server methods declared in webserver.h
+//
+// History:     Apr-18-2023         Rbergen     Created
+//---------------------------------------------------------------------------
+
+#include "globals.h"
+#include "webserver.h"
+
+namespace 
+{
+    bool IsParamTrue(AsyncWebParameter *param)
+    {
+        return param->value() == "true" || strtol(param->value().c_str(), NULL, 10);
+    }
+
+    bool IsPostParamTrue(AsyncWebServerRequest * pRequest, const String &paramName)
+    {
+        if (!pRequest->hasParam(paramName, true, false))
+            return false;
+        
+        debugV("found %s", paramName.c_str());
+
+        // If found, parse it and pass it off to the setter
+        return IsParamTrue(pRequest->getParam(paramName, true, false));
+    }
+
+    template<typename Tv>
+    using PostParamSetter = std::function<void(Tv)>;
+
+    template<typename Tv>
+    void PushPostParam(AsyncWebServerRequest * pRequest, const String &paramName, PostParamSetter<Tv> setter)
+    {
+        if (pRequest->hasParam(paramName, true, false))
+        {
+            debugV("found %s", paramName.c_str());
+
+            // If found, parse it and pass it off to the setter
+            setter(pRequest->getParam(paramName, true, false)->value());
+        }       
+    }
+
+    template<>
+    void PushPostParam<bool>(AsyncWebServerRequest * pRequest, const String &paramName, PostParamSetter<bool> setter)
+    {
+        if (pRequest->hasParam(paramName, true, false))
+        {
+            debugV("found %s", paramName.c_str());
+
+            // If found, parse it and pass it off to the setter
+            setter(IsParamTrue(pRequest->getParam(paramName, true, false)));
+        }       
+    }
+}
+
+void CWebServer::GetSettings(AsyncWebServerRequest * pRequest)
+{
+    debugV("GetSettings");
+
+    auto response = new AsyncJsonResponse(false, JSON_BUFFER_BASE_SIZE);
+    response->addHeader("Server","NightDriverStrip");
+    auto root = response->getRoot();
+    JsonObject jsonObject = root.to<JsonObject>();
+    
+    g_aptrDeviceConfig->SerializeToJSON(jsonObject);
+    jsonObject["effectInterval"] = g_aptrEffectManager->GetInterval();
+
+    response->setLength();

```

Comment:
Doesn't AddCORESHeaderAndSendOKResponse do this same workflow?

Reply:
Kind of. AddCORESHeaderAndSendOKResponse creates its own response, using an optional text parameter. This method sends back a JSON response that's created earlier. This is the same approach used by other CWebServer endpoint methods that send back content in their response.

That said, I guess overall a few lines of code could be saved if AddCORESHeaderAndSendOKResponse was overloaded to also accept a response object, and that overload was used by all methods that return a content response.